src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -170,9 +170,12 @@ public int MaxDepth
             {
                 VerifyMutable();
                 _maxDepth = value;
+                EffectiveMaxDepth = (value == 0 ? JsonReaderOptions.DefaultMaxDepth : value);
             }
         }
 
+        internal int EffectiveMaxDepth { get; private set; } = JsonReaderOptions.DefaultMaxDepth;
```

Comment:
> It appears the `Utf8JsonWriter` now has an upper limit of 1,000 so we would never get a `StackOverflowException`.

Why are we using the `JsonReaderOptions` `DefaultMaxDepth` of 64 to guard  against the depth while writing? Where does the 1,000 limit on the writer come into play here?

Reply:
> Why are we using the JsonReaderOptions DefaultMaxDepth of 64 to guard against the depth while writing

The default is 64 on the serializer only because that is the default the reader has, so using the same constant.

> Where does the 1,000 limit on the writer come into play here?

It doesn't really - you may or may not get a StackOverflowException before the failsafe 1,000 is hit.